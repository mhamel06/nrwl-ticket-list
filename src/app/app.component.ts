import { Component } from '@angular/core';
import { CodeAssignmentState } from './models/code-assignment-state.model';
import { Store } from '@ngrx/store';
import { TicketActions, UserActions } from './state/actions';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.css']
})
export class AppComponent {

  constructor(private store: Store<CodeAssignmentState>) { }

  ngOnInit() {
    this.store.dispatch(TicketActions.loadTickets());
    this.store.dispatch(UserActions.loadUsers());
  }
}
