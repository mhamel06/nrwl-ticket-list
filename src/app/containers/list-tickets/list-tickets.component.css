table {
    width: 100%;
}
.filter-row{
    width:100%;
}