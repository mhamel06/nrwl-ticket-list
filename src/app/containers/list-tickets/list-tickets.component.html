<p>
  list-tickets works!
</p>
<ul>
  <li *ngFor="let t of tickets$ | async">
  Ticket: {{t.id}}, {{t.description}}
  </li>
</ul>

