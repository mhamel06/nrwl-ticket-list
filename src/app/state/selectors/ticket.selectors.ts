import { ticketEntityAdapter } from '../reducers/ticket.reducer';
import { createFeatureSelector } from '@ngrx/store';
import { Ticket } from '../../backend.service';
import { EntityState } from '@ngrx/entity';
import { CodeAssignmentState } from '../../models/code-assignment-state.model';

export const TICKETS_FEATURE_KEY = 'tickets';


const getTicketState = createFeatureSelector<CodeAssignmentState, EntityState<Ticket>>(TICKETS_FEATURE_KEY);

const {
  selectEntities,
  selectAll,
} = ticketEntityAdapter.getSelectors(getTicketState);

export const selectAllTickets = selectAll;