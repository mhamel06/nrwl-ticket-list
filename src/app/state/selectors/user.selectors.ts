import { createFeatureSelector } from '@ngrx/store';
import { User } from '../../backend.service';
import { EntityState } from '@ngrx/entity';
import { CodeAssignmentState } from '../../models/code-assignment-state.model';
import { userEntityAdapter } from '../reducers/user.reducer';

export const USER_FEATURE_KEY = 'users';


const getUserState = createFeatureSelector<CodeAssignmentState, EntityState<User>>(USER_FEATURE_KEY);

const {
  selectEntities,
  selectAll,
} = userEntityAdapter.getSelectors(getUserState);

export const selectAllUsers = selectAll;
export const selectAllUserEntities = selectEntities;