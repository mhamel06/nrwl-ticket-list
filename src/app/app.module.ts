import { BrowserModule } from '@angular/platform-browser';
import { NgModule } from '@angular/core';
import { AppComponent } from './app.component';
import {BackendService} from './backend.service';
import { RouterModule } from '@angular/router';
import { EffectsModule } from '@ngrx/effects';
import { ListTicketsComponent } from './containers/list-tickets/list-tickets.component';
import { ViewTicketsComponent } from './containers/view-tickets/view-tickets.component';
import { StoreModule } from '@ngrx/store';
import { TicketEffects } from './state/effects/ticket.effects';
import { CodeAssignmentReducers } from './state/reducers';
import { StoreDevtoolsModule } from '@ngrx/store-devtools';
import { ticketsEntityReducer } from './state/reducers/ticket.reducer';

@NgModule({
  declarations: [
    AppComponent,
    ListTicketsComponent,
    ViewTicketsComponent
  ],
  imports: [
    BrowserModule,
    StoreModule.forRoot({
      tickets: ticketsEntityReducer
    }),
    StoreDevtoolsModule.instrument(),
    RouterModule.forRoot([
      {path: 'tickets', component: ListTicketsComponent},
      {path: 'tickets/{id}', component: ListTicketsComponent},
      {path: '', redirectTo: '/tickets', pathMatch:'full'}
    ]),
    EffectsModule.forRoot([
      TicketEffects
    ]),
  ],
  providers: [BackendService],
  bootstrap: [AppComponent]
})
export class AppModule {

}
