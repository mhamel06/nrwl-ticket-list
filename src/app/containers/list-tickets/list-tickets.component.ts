import { Component, OnInit } from '@angular/core';
import { select, Store } from '@ngrx/store';
import { CodeAssignmentState } from '../../models/code-assignment-state.model';
import { TicketActions } from '../../state/actions';
import { Observable } from 'rxjs';
import { Ticket } from '../../backend.service';
import { selectAllTickets } from '../../state/selectors/ticket.selectors';

@Component({
  selector: 'app-list-tickets',
  templateUrl: './list-tickets.component.html',
  styleUrls: ['./list-tickets.component.css']
})
export class ListTicketsComponent implements OnInit {

  tickets$: Observable<Ticket[]>;

  constructor(private store: Store<CodeAssignmentState>) { }

  ngOnInit() {
    this.store.dispatch(TicketActions.loadTickets());

    this.tickets$ = this.store.pipe(
      select(selectAllTickets)
    );
  }
}
