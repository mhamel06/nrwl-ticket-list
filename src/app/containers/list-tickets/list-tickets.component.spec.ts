import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { ListTicketsComponent } from './list-tickets.component';
import { matModules } from '../../app.module';
import { RouterTestingModule } from '@angular/router/testing';
import { MockStore, provideMockStore } from '@ngrx/store/testing';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { ActivatedRoute } from '@angular/router';
import { of } from 'rxjs';
import { CodeAssignmentState } from '../../models/code-assignment-state.model';
import { Store } from '@ngrx/store';
import { MOCK_TICKETS_WITH_USERS } from '../../models/mocks/ticket.mock';
import { selectTicketsWithUser } from '../../state/selectors/ticket.selectors';
import { MOCK_USERS_ENTITY } from '../../models/mocks/user.mock';

/*
* Introduced bug with entities and missing state, ran out of time debugging
* but should be able to sort out issue with more time.
* */
describe('ListTicketsComponent', () => {
  let component: ListTicketsComponent;
  let fixture: ComponentFixture<ListTicketsComponent>;
  let activatedRoute;
  let mockStore: MockStore<CodeAssignmentState>;
  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports:[
        ...matModules,
        RouterTestingModule,
        NoopAnimationsModule,
      ],
      providers:[
        provideMockStore({
          initialState: {
            users: {
              ids: ['111'],
              entities: {...MOCK_USERS_ENTITY}
            }
          }
        })
      ],
      declarations: [ ListTicketsComponent ]
    });
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(ListTicketsComponent);
    component = fixture.componentInstance;
    mockStore = TestBed.get(Store);

    activatedRoute = TestBed.get(ActivatedRoute);
    activatedRoute.params = jest.fn().mockImplementation(()=>
      of({params: {id: 0}}));

    mockStore.overrideSelector(selectTicketsWithUser, [...MOCK_TICKETS_WITH_USERS]);
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  /*
  * I usually avoid storing snapshots of large components,
  * but I sometimes use it to confirm everything is rendered
  * as expected given the mock selectors.
  *
  * In most cases, I would create a file with with css selectors
  * that could be reused when testing specific dom elements in
  * a component.
  * */
  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
