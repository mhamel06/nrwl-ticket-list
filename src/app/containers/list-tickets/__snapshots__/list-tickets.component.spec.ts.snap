// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListTicketsComponent should match snapshot 1`] = `
<app-list-tickets
  dataSource={[Function MatTableDataSource]}
  displayedColumns={[Function Array]}
  onDestroy$={[Function Subject]}
  store={[Function MockStore]}
  tickets$={[Function Store]}
  users$={[Function Store]}
>
  <mat-form-field
    class="mat-form-field ng-tns-c6-0 mat-primary mat-form-field-type-mat-input mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-has-label mat-form-field-hide-placeholder _mat-animation-noopable"
  >
    <div
      class="mat-form-field-wrapper"
    >
      <div
        class="mat-form-field-flex"
      >
        
        
        <div
          class="mat-form-field-infix"
        >
          <input
            aria-invalid="false"
            aria-required="false"
            class="mat-input-element mat-form-field-autofill-control cdk-text-field-autofill-monitored"
            id="mat-input-2"
            matinput=""
            ng-reflect-placeholder="Ticket Description"
            placeholder="Ticket Description"
          />
          <span
            class="mat-form-field-label-wrapper"
          >
            
            <label
              aria-owns="mat-input-2"
              class="mat-form-field-label ng-tns-c6-0 mat-empty mat-form-field-empty ng-star-inserted"
              for="mat-input-2"
              id="mat-form-field-label-13"
              ng-reflect-disabled="true"
              ng-reflect-ng-switch="false"
            >
              
              
              <span
                class="ng-tns-c6-0 ng-star-inserted"
              >
                Ticket Description
              </span>
              
              
            </label>
          </span>
        </div>
        
      </div>
      
      <div
        class="mat-form-field-underline ng-tns-c6-0 ng-star-inserted"
      >
        <span
          class="mat-form-field-ripple"
        />
      </div>
      <div
        class="mat-form-field-subscript-wrapper"
        ng-reflect-ng-switch="hint"
      >
        
        
        <div
          class="mat-form-field-hint-wrapper ng-tns-c6-0 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
          style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
        >
          
          <div
            class="mat-form-field-hint-spacer"
          />
        </div>
      </div>
    </div>
  </mat-form-field><button
    class="mat-button _mat-animation-noopable"
    mat-button=""
  >
    <span
      class="mat-button-wrapper"
    >
      New Ticket
    </span>
    <div
      class="mat-button-ripple mat-ripple"
      matripple=""
      ng-reflect-centered="false"
      ng-reflect-disabled="false"
      ng-reflect-trigger="[object HTMLButtonElement]"
    />
    <div
      class="mat-button-focus-overlay"
    />
  </button><div
    class="filter-row"
  >
    <mat-form-field
      class="filter-row mat-form-field ng-tns-c6-1 mat-primary mat-form-field-type-mat-input mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-has-label mat-form-field-hide-placeholder _mat-animation-noopable"
    >
      <div
        class="mat-form-field-wrapper"
      >
        <div
          class="mat-form-field-flex"
        >
          
          
          <div
            class="mat-form-field-infix"
          >
            <input
              aria-invalid="false"
              aria-required="false"
              class="mat-input-element mat-form-field-autofill-control cdk-text-field-autofill-monitored"
              id="mat-input-3"
              matinput=""
              ng-reflect-placeholder="Filter"
              placeholder="Filter"
            />
            <span
              class="mat-form-field-label-wrapper"
            >
              
              <label
                aria-owns="mat-input-3"
                class="mat-form-field-label ng-tns-c6-1 mat-empty mat-form-field-empty ng-star-inserted"
                for="mat-input-3"
                id="mat-form-field-label-15"
                ng-reflect-disabled="true"
                ng-reflect-ng-switch="false"
              >
                
                
                <span
                  class="ng-tns-c6-1 ng-star-inserted"
                >
                  Filter
                </span>
                
                
              </label>
            </span>
          </div>
          
        </div>
        
        <div
          class="mat-form-field-underline ng-tns-c6-1 ng-star-inserted"
        >
          <span
            class="mat-form-field-ripple"
          />
        </div>
        <div
          class="mat-form-field-subscript-wrapper"
          ng-reflect-ng-switch="hint"
        >
          
          
          <div
            class="mat-form-field-hint-wrapper ng-tns-c6-1 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
            style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
          >
            
            <div
              class="mat-form-field-hint-spacer"
            />
          </div>
        </div>
      </div>
    </mat-form-field>
  </div><table
    class="mat-elevation-z0 mat-table"
    mat-table=""
    ng-reflect-data-source="[object Object]"
    role="grid"
  >
    <thead
      role="rowgroup"
    >
      
      <tr
        class="mat-header-row ng-star-inserted"
        mat-header-row=""
        role="row"
      >
        
        <th
          class="mat-header-cell cdk-column-id mat-column-id ng-star-inserted"
          mat-header-cell=""
          role="columnheader"
        >
           ID 
        </th>
        <th
          class="mat-header-cell cdk-column-description mat-column-description ng-star-inserted"
          mat-header-cell=""
          role="columnheader"
        >
           Description 
        </th>
        <th
          class="mat-header-cell cdk-column-assigneeName mat-column-assigneeName ng-star-inserted"
          mat-header-cell=""
          role="columnheader"
        >
           Assignee 
        </th>
        <th
          class="mat-header-cell cdk-column-completed mat-column-completed ng-star-inserted"
          mat-header-cell=""
          role="columnheader"
        >
           Completed 
        </th>
        <th
          class="mat-header-cell cdk-column-view mat-column-view ng-star-inserted"
          mat-header-cell=""
          role="columnheader"
        />
      </tr>
    </thead>
    <tbody
      role="rowgroup"
    >
      
      <tr
        class="mat-row ng-star-inserted"
        mat-row=""
        role="row"
      >
        
        <td
          class="mat-cell cdk-column-id mat-column-id ng-star-inserted"
          mat-cell=""
          role="gridcell"
        >
           0 
        </td>
        <td
          class="mat-cell cdk-column-description mat-column-description ng-star-inserted"
          mat-cell=""
          role="gridcell"
        >
           Install a monitor arm 
        </td>
        <td
          class="mat-cell cdk-column-assigneeName mat-column-assigneeName ng-star-inserted"
          mat-cell=""
          role="gridcell"
        >
          <mat-form-field
            class="mat-form-field ng-tns-c6-2 mat-primary mat-form-field-type-mat-select mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-hide-placeholder _mat-animation-noopable"
          >
            <div
              class="mat-form-field-wrapper"
            >
              <div
                class="mat-form-field-flex"
              >
                
                
                <div
                  class="mat-form-field-infix"
                >
                  <mat-select
                    aria-disabled="false"
                    aria-invalid="false"
                    aria-multiselectable="false"
                    aria-required="false"
                    class="mat-select ng-tns-c9-3 mat-select-empty ng-star-inserted"
                    id="mat-select-4"
                    ng-reflect-value="111"
                    role="listbox"
                    tabindex="0"
                  >
                    <div
                      aria-hidden="true"
                      cdk-overlay-origin=""
                      class="mat-select-trigger"
                    >
                      <div
                        class="mat-select-value"
                        ng-reflect-ng-switch="true"
                      >
                        
                        <span
                          class="mat-select-placeholder ng-tns-c9-3 ng-star-inserted"
                        >
                          Â 
                        </span>
                        
                      </div>
                      <div
                        class="mat-select-arrow-wrapper"
                      >
                        <div
                          class="mat-select-arrow"
                        />
                      </div>
                    </div>
                    
                  </mat-select>
                  <span
                    class="mat-form-field-label-wrapper"
                  >
                    
                  </span>
                </div>
                
              </div>
              
              <div
                class="mat-form-field-underline ng-tns-c6-2 ng-star-inserted"
              >
                <span
                  class="mat-form-field-ripple"
                />
              </div>
              <div
                class="mat-form-field-subscript-wrapper"
                ng-reflect-ng-switch="hint"
              >
                
                
                <div
                  class="mat-form-field-hint-wrapper ng-tns-c6-2 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
                  style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
                >
                  
                  <div
                    class="mat-form-field-hint-spacer"
                  />
                </div>
              </div>
            </div>
          </mat-form-field>
        </td>
        <td
          class="mat-cell cdk-column-completed mat-column-completed ng-star-inserted"
          mat-cell=""
          role="gridcell"
        >
          <mat-form-field
            class="mat-form-field ng-tns-c6-4 mat-primary mat-form-field-type-mat-select mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-hide-placeholder _mat-animation-noopable"
          >
            <div
              class="mat-form-field-wrapper"
            >
              <div
                class="mat-form-field-flex"
              >
                
                
                <div
                  class="mat-form-field-infix"
                >
                  <mat-select
                    aria-disabled="false"
                    aria-invalid="false"
                    aria-multiselectable="false"
                    aria-required="false"
                    class="mat-select ng-tns-c9-5 mat-select-empty ng-star-inserted"
                    id="mat-select-5"
                    ng-reflect-value="false"
                    role="listbox"
                    tabindex="0"
                  >
                    <div
                      aria-hidden="true"
                      cdk-overlay-origin=""
                      class="mat-select-trigger"
                    >
                      <div
                        class="mat-select-value"
                        ng-reflect-ng-switch="true"
                      >
                        
                        <span
                          class="mat-select-placeholder ng-tns-c9-5 ng-star-inserted"
                        >
                          Â 
                        </span>
                        
                      </div>
                      <div
                        class="mat-select-arrow-wrapper"
                      >
                        <div
                          class="mat-select-arrow"
                        />
                      </div>
                    </div>
                    
                  </mat-select>
                  <span
                    class="mat-form-field-label-wrapper"
                  >
                    
                  </span>
                </div>
                
              </div>
              
              <div
                class="mat-form-field-underline ng-tns-c6-4 ng-star-inserted"
              >
                <span
                  class="mat-form-field-ripple"
                />
              </div>
              <div
                class="mat-form-field-subscript-wrapper"
                ng-reflect-ng-switch="hint"
              >
                
                
                <div
                  class="mat-form-field-hint-wrapper ng-tns-c6-4 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
                  style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
                >
                  
                  <div
                    class="mat-form-field-hint-spacer"
                  />
                </div>
              </div>
            </div>
          </mat-form-field>
        </td>
        <td
          class="mat-cell cdk-column-view mat-column-view ng-star-inserted"
          mat-cell=""
          role="gridcell"
        >
          <a
            href="/0"
            ng-reflect-router-link="0"
          >
            View
          </a>
        </td>
      </tr>
      <tr
        class="mat-row ng-star-inserted"
        mat-row=""
        role="row"
      >
        
        <td
          class="mat-cell cdk-column-id mat-column-id ng-star-inserted"
          mat-cell=""
          role="gridcell"
        >
           1 
        </td>
        <td
          class="mat-cell cdk-column-description mat-column-description ng-star-inserted"
          mat-cell=""
          role="gridcell"
        >
           Move the desk to the new location 
        </td>
        <td
          class="mat-cell cdk-column-assigneeName mat-column-assigneeName ng-star-inserted"
          mat-cell=""
          role="gridcell"
        >
          <mat-form-field
            class="mat-form-field ng-tns-c6-6 mat-primary mat-form-field-type-mat-select mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-hide-placeholder _mat-animation-noopable"
          >
            <div
              class="mat-form-field-wrapper"
            >
              <div
                class="mat-form-field-flex"
              >
                
                
                <div
                  class="mat-form-field-infix"
                >
                  <mat-select
                    aria-disabled="false"
                    aria-invalid="false"
                    aria-multiselectable="false"
                    aria-required="false"
                    class="mat-select ng-tns-c9-7 mat-select-empty ng-star-inserted"
                    id="mat-select-6"
                    ng-reflect-value="111"
                    role="listbox"
                    tabindex="0"
                  >
                    <div
                      aria-hidden="true"
                      cdk-overlay-origin=""
                      class="mat-select-trigger"
                    >
                      <div
                        class="mat-select-value"
                        ng-reflect-ng-switch="true"
                      >
                        
                        <span
                          class="mat-select-placeholder ng-tns-c9-7 ng-star-inserted"
                        >
                          Â 
                        </span>
                        
                      </div>
                      <div
                        class="mat-select-arrow-wrapper"
                      >
                        <div
                          class="mat-select-arrow"
                        />
                      </div>
                    </div>
                    
                  </mat-select>
                  <span
                    class="mat-form-field-label-wrapper"
                  >
                    
                  </span>
                </div>
                
              </div>
              
              <div
                class="mat-form-field-underline ng-tns-c6-6 ng-star-inserted"
              >
                <span
                  class="mat-form-field-ripple"
                />
              </div>
              <div
                class="mat-form-field-subscript-wrapper"
                ng-reflect-ng-switch="hint"
              >
                
                
                <div
                  class="mat-form-field-hint-wrapper ng-tns-c6-6 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
                  style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
                >
                  
                  <div
                    class="mat-form-field-hint-spacer"
                  />
                </div>
              </div>
            </div>
          </mat-form-field>
        </td>
        <td
          class="mat-cell cdk-column-completed mat-column-completed ng-star-inserted"
          mat-cell=""
          role="gridcell"
        >
          <mat-form-field
            class="mat-form-field ng-tns-c6-8 mat-primary mat-form-field-type-mat-select mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-hide-placeholder _mat-animation-noopable"
          >
            <div
              class="mat-form-field-wrapper"
            >
              <div
                class="mat-form-field-flex"
              >
                
                
                <div
                  class="mat-form-field-infix"
                >
                  <mat-select
                    aria-disabled="false"
                    aria-invalid="false"
                    aria-multiselectable="false"
                    aria-required="false"
                    class="mat-select ng-tns-c9-9 mat-select-empty ng-star-inserted"
                    id="mat-select-7"
                    ng-reflect-value="false"
                    role="listbox"
                    tabindex="0"
                  >
                    <div
                      aria-hidden="true"
                      cdk-overlay-origin=""
                      class="mat-select-trigger"
                    >
                      <div
                        class="mat-select-value"
                        ng-reflect-ng-switch="true"
                      >
                        
                        <span
                          class="mat-select-placeholder ng-tns-c9-9 ng-star-inserted"
                        >
                          Â 
                        </span>
                        
                      </div>
                      <div
                        class="mat-select-arrow-wrapper"
                      >
                        <div
                          class="mat-select-arrow"
                        />
                      </div>
                    </div>
                    
                  </mat-select>
                  <span
                    class="mat-form-field-label-wrapper"
                  >
                    
                  </span>
                </div>
                
              </div>
              
              <div
                class="mat-form-field-underline ng-tns-c6-8 ng-star-inserted"
              >
                <span
                  class="mat-form-field-ripple"
                />
              </div>
              <div
                class="mat-form-field-subscript-wrapper"
                ng-reflect-ng-switch="hint"
              >
                
                
                <div
                  class="mat-form-field-hint-wrapper ng-tns-c6-8 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
                  style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
                >
                  
                  <div
                    class="mat-form-field-hint-spacer"
                  />
                </div>
              </div>
            </div>
          </mat-form-field>
        </td>
        <td
          class="mat-cell cdk-column-view mat-column-view ng-star-inserted"
          mat-cell=""
          role="gridcell"
        >
          <a
            href="/1"
            ng-reflect-router-link="1"
          >
            View
          </a>
        </td>
      </tr>
    </tbody>
    <tfoot
      class="mat-table-sticky"
      role="rowgroup"
      style="display: none; bottom: 0px; position: sticky; z-index: 10;"
    >
      
    </tfoot>
  </table>
</app-list-tickets>
`;
